### PR TITLE
feat: header change when logged in or not

### PR DESCRIPTION
**Pull request by NyAndoMayah**

### CODE REVIEW

File: src/layout/AuthenticatedHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            icon="material-symbols-light:login"
+            className="text-2xl"
+          />
+        </Button>
+      </div>
+    </div>
+  );
+};
+
+export const AuthenticatedHeader = memo(AuthenticatedHeaderComponent);
```
Comment (YumeT023) [inline src/layout/AuthenticatedHeader.tsx]: why are you memoizing this component ?

File: src/layout/AuthenticatedHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+} from "@/components/shadcn-ui/avatar";
+import {
+  NavigationMenu,
+  NavigationMenuItem,
+  NavigationMenuLink,
+  NavigationMenuList,
+  navigationMenuTriggerStyle,
+} from "@/components/shadcn-ui/navigation-menu";
+
+export const AuthenticatedHeaderComponent: FC = () => {
```
Comment (YumeT023) [inline src/layout/AuthenticatedHeader.tsx]: Move this component and AnonymousHeader inside the Navbar component file
DO not export them, only export the Navbar

File: src/layout/AuthenticatedHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                  data-testid="about-menu"
+                  className={navigationMenuTriggerStyle()}
+                >
+                  Posts
+                </NavigationMenuLink>
+              </Link>
+            </NavigationMenuItem>
+            <NavigationMenuItem className="mx-4 w-40">
+              {/* TODO: change this to the profile link */}
+              <Link to="/">
```
Comment (YumeT023) [inline src/layout/AuthenticatedHeader.tsx]: update this profile path

**Review by YumeT023 — CHANGES_REQUESTED** [review]

File: src/layout/AuthenticatedHeader.tsx (hunk 1 above)
Comment (NyAndoMayah) [reply src/layout/AuthenticatedHeader.tsx]: Because you memoized unauthenticatedHeader?

Comment (YumeT023) [reply src/layout/AuthenticatedHeader.tsx]: Here AuthenticatedHeader just render jsx, no prop depends on it,
But yeah, we do access this component everytime in the app so leave it like so

**Pull request merged**